packages/octotiger/package.py

```diff
@@ -99,7 +99,7 @@ class Octotiger(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('boost@1.61.0: cxxstd=14', when="@0.8.0")
     depends_on('boost@1.74.0: cxxstd=17', when="@0.9.0:")
     depends_on('hdf5 +threadsafe +szip +hl -mpi ')
-    depends_on('silo@4.10.2-bsd:4.11-bsd ')
+    depends_on('silo@4.10.2-bsd+hdf5:4.11-bsd+hdf5 ')
```

Comment:
Interesting! The hdf5 variant of silo should be enabled by default -- though I guess if there is already a version installed without this variant the spack concretizer might pick it! So we can certainly add the constraint.

However, putting the hdf5 variant here breaks the version range syntax! I'd suggest to simply add this with another depends_on statement:

```suggestion
    depends_on('silo@4.10.2-bsd:4.11-bsd ')
    depends_on('silo+hdf5 ')
```